### ISSUE TITLE
New: rule tests-ordering

### ISSUE THREAD

**Issue by Aladdin-ADD** [not-an-aardvark/eslint-plugin-eslint-plugin#53]
if a rule has a very large number of tests cases, it would be fine to reorder them based on `options`.

this rule can enforce code like this to improve readability:
```js
      new RuleTester().run('foo', bar, {
        valid: [

          // valid tests for options: ['foo']
          { code: 'codeA', options: ['foo'] },
          { code: 'codeB', options: ['foo'] },
          // and more...

          // valid tests for options: ['bar']
          { code: 'codeA', options: ['bar'] },
          { code: 'codeB', options: ['bar'] },
          // and more...
        ],
      });

```

**Comment by not-an-aardvark** [comment]
How would ordering be determined? For example, does `options: ['foo']` come before `options: ['foo', 'bar']`?

**Comment by Aladdin-ADD** [comment]
it aims to put the tests having same options together, but it's fine to add an option to enforce options ordering alphabetically:
```js
// tests for options: []
...

// tests for options: ['bar']
...

// tests for options: ['bar', 'foo']
...

// tests for options: ['foo']
...

```

**Comment by not-an-aardvark** [comment]
Sounds good to me. I don't think it's necessary to make the options alphabetical -- that seems like it could be confusing.
